### DISCUSSION TITLE
Is it possible to cancel/delete a scheduled event?

### DISCUSSION THREAD

**Discussion by fkneist** [etf1/kafka-message-scheduler#46] [Q&A]
Let's say I send an event to kafka-message-scheduler at 10:00 because I want to trigger sending an email at 16:00. Meanwhile, circumstances change and at 13:00 I know, "I don't want to send that email anymore", is there a way to "cancel" or delete a scheduled event then?

**Comment by fkarakas** [comment]
Hello @fkneist,

Yes it is possible to delete a schedule not yet triggered, what to have to do is to send a schedule withe corresponding schedule-id with a nil payload (Value is nil), this is a tombstone message and it will remove the schedule from the scheduler, kafka will also delete all the messages with this key when compaction will occur.

For more information you can check the client lib:
https://github.com/etf1/kafka-message-scheduler/tree/main/clientlib#to-delete-a-scheduled-message

https://github.com/etf1/kafka-message-scheduler/blob/9b44c3c8668f117c243678f5974eabf03fd6d174/clientlib/clientlib.go#L144

Regards,

**Reply by fkneist** [reply]
Thank you for clarifying!
